rust_src/src/frames.rs

```diff
@@ -195,6 +208,115 @@ impl From<LispFrameLiveOrSelected> for LispFrameRef {
     }
 }
 
+// Perform the switch to frame FRAME.
+//
+// If FRAME is a switch-frame event `(switch-frame FRAME1)', use
+// FRAME1 as frame.
+//
+// If TRACK is non-zero and the frame that currently has the focus
+// redirects its focus to the selected frame, redirect that focused
+// frame's focus to FRAME instead.
+//
+// FOR_DELETION non-zero means that the selected frame is being
+// deleted, which includes the possibility that the frame's terminal
+// is dead.
+//
+// The value of NORECORD is passed as argument to Fselect_window.
+#[no_mangle]
+pub extern "C" fn do_switch_frame(
+    mut frame: LispObject,
+    track: i32,
```

Comment:
It looks like `track` is really `bool`. Does that work?

Reply:
seems working simply with bools